compiler/mir/mirgen.nim

```diff
@@ -1728,6 +1728,11 @@ proc genx(c: var TCtx, e: PMirExpr, i: int) =
   template recurse() =
     genx(c, e, i - 1)
 
+  proc pick(typ: PType, mut, nonMut: MirNodeKind): MirNodeKind =
+    case typ.skipTypes(abstractInst).kind
+    of tyVar: mut
+    else:     nonMut
```

Comment:
```suggestion
  proc viewKind(typ: PType, kind: MirNodeKind): MirNodeKind =
    proc pick(typ: PType, mut, nonMut: MirNodeKind): MirNodeKind =
      case typ.skipTypes(abstractInst).kind
      of tyVar: mut
      else:     nonMut

    case kind
    of mnkView: pick(typ, mnkMutView, mnkView)
    of mnkToSlice: pick(typ, mnkMutToSlice, mnkToSlice)
    else: unreachable(kind)
```

That might be cleaned up a tad more perhaps, but this way the mut vs non-mut decision is in one place (harder to accidentally swap param positions). Might not be worth it, just something to consider.

Reply:
Yeah, problems arising from accidentally swapping the parameters is something I've also thought about. I'll look into how much effort it'd take to integrate this into the PMIR instead -- if too much, I'd use your suggestion.